examples/context7-rs/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+#![allow(clippy::all)]
+
+#[allow(warnings)]
+mod bindings;
+
+use bindings::exports::local::context7::context7::{DocsResponse, Guest, LibraryResult, SearchResponse};
+use bindings::wasi::cli::environment;
+use bindings::wasi::http::outgoing_handler;
+use bindings::wasi::http::types::{self, Fields, IncomingResponse, Method, OutgoingRequest, Scheme};
+use bindings::wasi::io::streams::StreamError;
+use serde_json::Value;
+use urlencoding::encode;
+
+const CONTEXT7_HOST: &str = "context7.com";
+const BASE_PATH: &str = "/api";
+const DEFAULT_MINIMUM_TOKENS: u32 = 10_000;
+const DEFAULT_USER_AGENT: &str = "context7-wasm-component/1.0.0";
+const NO_CONTENT_MESSAGES: [&str; 2] = ["No content available", "No context data available"];
+
+struct Context7;
+
+impl Guest for Context7 {
+    fn resolve_library_id(library_name: String) -> SearchResponse {
+        let trimmed = library_name.trim();
+        if trimmed.is_empty() {
+            return SearchResponse {
+                success: false,
+                results: Vec::new(),
+                error: Some("Library name is required".to_string()),
+            };
+        }
+
+        let encoded_query = encode(trimmed);
+        let endpoint = format!("/v1/search?query={encoded_query}");
+        match make_request(Method::Get, &endpoint, None, get_api_key().as_deref()) {
+            Ok(body) => match parse_search_response(&body) {
+                Ok(response) => response,
+                Err(message) => SearchResponse {
+                    success: false,
+                    results: Vec::new(),
+                    error: Some(message),
+                },
+            },
+            Err(message) => SearchResponse {
+                success: false,
+                results: Vec::new(),
+                error: Some(message),
+            },
+        }
+    }
+
+    fn get_library_docs(
+        context7_compatible_library_id: String,
+        topic: Option<String>,
+        tokens: Option<u32>,
+    ) -> DocsResponse {
+        let trimmed = context7_compatible_library_id.trim();
+        if trimmed.is_empty() {
+            return DocsResponse {
+                success: false,
+                content: None,
+                error: Some("Context7 compatible library ID is required".to_string()),
+            };
+        }
+
+        let library_id = trimmed.trim_start_matches('/').to_string();
+        if library_id.is_empty() {
+            return DocsResponse {
+                success: false,
+                content: None,
+                error: Some("Context7 compatible library ID is required".to_string()),
+            };
+        }
+
+        let token_count = tokens
+            .map(|value| value.max(DEFAULT_MINIMUM_TOKENS))
+            .unwrap_or(DEFAULT_MINIMUM_TOKENS);
+
+        let mut parameters = vec![format!("tokens={token_count}"), "type=txt".to_string()];
+
+        if let Some(topic_value) = topic.as_ref().map(|value| value.trim()).filter(|v| !v.is_empty()) {
```

Comment:
Nit: Has this had `cargo fmt` run on it? I think there are several of these things that can be formatted cleaner